src/scripts/migrate.py

```diff
@@ -0,0 +1,141 @@
+from oaklib.resource import OntologyResource
+
+
+# TODO: implement this func:
+# todo: IDs should be int or str? prolly str
+def determine_next_available_mondo_id(min_id: str, mondo_termlist_df: pd.DataFrame) -> str:
+    """Starting from `min_id`, count up and check until finding the next ID."""
+    next_id = str(0)
+    return next_id
+
+
+def _get_direct_parents(curie: str, ontology: ProntoImplementation):
+    """Get CURIEs of parents of a class
+    todo: `ontology` typing: should probably be typed agnostic of the implementation. But it inherits from so
+     much stuff, I'm not sure which if any of these to use. I was expecting `BasicOntologyInterface`, but it doesn't
+     inherit from that: RelationGraphInterface, OboGraphInterface, ValidatorInterface, SearchInterface,
+     SubsetterInterface, MappingProviderInterface, PatcherInterface, SemanticSimilarityInterface, MetadataInterface,
+     DifferInterface, ABC
+    """
+    # TODO: How to get just the *direct* parents? (a) func for that? (b) look at relationships and see if subClass?
+    # rels = ontology.get_outgoing_relationships_by_curie(curie)  # works for ProntoImplementation
+    rels = ontology.relationships([curie])
+    p = []
+    for rel, parents in rels.items():
+        # print(f'  {rel} ! {ontology.get_label_by_curie(rel)}')
+        for parent in parents:
+            # print(f'    {parent} ! {ontology.get_label_by_curie(parent)}')
+            p.append(parent)
+    return p
+
+
+def _get_all_term_curies(ontology: ProntoImplementation, is_omim=False) -> List[str]:
+    """Get all terms as CURIEs
+    todo: Will we need prefix_map from our config `.yml`s? Or can OAK handle automatically?
+     - reasons we might need: (i) filter correct terms, (ii) convert URI to CURIE if get URI
+     """
+    # todo: if oak can't handle automatically, I think we'll need `prefix_map`:
+    terms = [x for x in ontology.entities()]
+    # todo: temp until I figure out if (a) we need `prefix_map`, or (b) OAK can somehow handle automatically:
+    if is_omim:
+        terms = [x for x in terms if any([x.startswith(y) for y in [
+            'OMIM:', 'OMIMPS:', 'https://omim.org/entry/', 'https://omim.org/phenotypicSeries/PS']])]
+        # prefix_map = {
+        #    'OMIM': 'https://omim.org/entry/',
+        #    'OMIMPS': 'https://omim.org/phenotypicSeries/PS',
+        # }
+        # todo: convert to CURIE using charlie's new lib
+    return terms
+
+
+# todo: temp links:
+# https://incatools.github.io/ontology-access-kit/
+# https://incatools.github.io/ontology-access-kit/intro/tutorial02.html
+def run(ontology_path: str, sssom_map_path: str, min_id: str, mondo_terms_path: str, outpath: str) -> pd.DataFrame:
+    """Run slurp pipeline for given ontology"""
+    # Read source files
+    # todo: `ontology`: If trouble, can try `SparqlImplementation`, but ~6 min to load and queries slow (cuz rdflib)
+    ontology = ProntoImplementation(OntologyResource(slug=ontology_path, local=True))
```

Comment:
## Success cases
I tried several other implementations first, but `ProntoImplementation`, which is what the documentation examples use, worked the best. `SparqlImplementation` mostly worked too, but it was _very_ slow (because of `rdflib`).

I experienced a lot of issues trying out the various implementations, and I created/updated some OAK GitHub issues.

## Failed cases
### `SqlImplementation`
I encountered 4 distinct issues, but they all seemed related to actually loading the ontology. I think it was related to an existing issue you opened, so I updated that: https://github.com/INCATools/ontology-access-kit/issues/90#issuecomment-1205957717

### `FunOwlImplementation`
I got no entities back when I queried for them: https://github.com/INCATools/ontology-access-kit/issues/218

## Other attempts
I didn't spend much time on these but I wanted to give them a shot.
- `ObographImplementation`: `NotImplementedError`
- `Bioportal`: `NotImplementedError`
- `BasicOntologyInterface`: `NotImplementedError` (I guess not surprising; it's an interface)

---

I looked at [the CLI](https://incatools.github.io/ontology-access-kit/cli.html) and I think it infers what implementation you need based on what you pass. Doesn't seem to be explicit. Maybe it is easier to use, but I'd prefer to use Python directly if I can.

Reply:
THIS IS GREAT! Thanks for doing this thorough analysis, I highly highly appreciated. This is long-game stuff and really valuable!

Reply:
Go ahead as you see fit.

Reply:
Thanks. It did take a fair amount of time to try these out. Glad that it was useful to do so. I'm going to likely pick `ProntoImplementation` or `SqlImplementation`. I do have some questions you might be able to answer regarding loading an ontology into a SQLite DB. I have a note to talk about that at one of our meetings.